src/prop/minisat/core/Solver.cc

```diff
@@ -428,7 +428,10 @@ bool Solver::addClause_(vec<Lit>& ps, bool removable, ClauseId& id)
             }
           }
           return ok;
-        } else return ok;
+        } else {
+          PROOF(id = ClauseIdUndef;);
```

Comment:
Can you update the bug w/ a more complete stack trace? I am not sure why setting the id for the calling context helps.